azurerm/resource_arm_public_ip.go

```diff
@@ -122,6 +135,13 @@ func resourceArmPublicIpCreate(d *schema.ResourceData, meta interface{}) error {
 
 	idleTimeout := d.Get("idle_timeout_in_minutes").(int)
 	ipAllocationMethod := network.IPAllocationMethod(d.Get("public_ip_address_allocation").(string))
+	ipVersion := network.IPVersion(d.Get("ip_version").(string))
+
+	if strings.ToLower(string(ipVersion)) == "ipv6" {
+		if strings.ToLower(string(ipAllocationMethod)) == "static" {
+			return fmt.Errorf("Cannot specify publicIpAllocationMethod as Static for IPv6 PublicIp")
```

Comment:
This should be moved [to a CustomizeDiff](https://github.com/terraform-providers/terraform-provider-azurerm/blob/cadce839c18ec0d88c9152c029108e350d4892b7/azurerm/resource_arm_servicebus_namespace.go#L94)

Reply:
since this field could be interpolated - I'd suggest we don't do that in this instance